### DISCUSSION TITLE
navigator.clipboard.writeText is not working.

### DISCUSSION THREAD

**Discussion by usuall** [zauberzeug/nicegui#1816] [Q&A] [answered]
### Question

I'm testing the Navigator.clipboard.writeText function.
If a sentence contains an open line character, it will not execute normally.
How do you solve it?

```py
@logger.catch
async def copy_SMS() -> None:
    sms ="TEST1\nTEST2\nTEST3"

    await ui.run_javascript(
        f'navigator.clipboard.writeText("{sms}")',
        respond=False,
    )
    ui.notify(
        message=f"SMS copied(Clipboard)\n",
        position="top-right",
        color="green",
    )

ui.run(title='test app', reload=False, native=True, port=8200, window_size=(1000, 900))
```

**Comment by falkoschindler** [comment, marked as answer]
In JavaScript you should use backticks for multi-line strings:
```py
async def copy_SMS() -> None:
    sms = 'TEST1\nTEST2'
    await ui.run_javascript(f'navigator.clipboard.writeText(`{sms}`)', respond=False)
```

**Reply by usuall** [reply]
@falkoschindler  
Thank you.

**Comment by natankeddem** [comment]
Also, be aware that clipboard manipulation might have some issues if you aren't utilizing a secure connection on some browsers.
https://stackoverflow.com/questions/51805395/navigator-clipboard-is-undefined
